src/generators/utils/parse.ts

```diff
@@ -65,33 +65,62 @@ function getDomain(url: string) {
     }
 }
 
+const radix = " !#$%&'()*+,-./0123456789:;=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+
+function encodeNumber(n: number, length: number | null) {
+    let s = '';
+    for (let c = n; c; c = Math.floor(c / radix.length)) {
+        s = radix[c % radix.length] + s;
+    }
+    if (length !== null) {
+        s = radix[0].repeat(length - s.length) + s;
+    }
+    return s;
+}
```

Comment:
Isn't it efficient to use the full-range utf-8 by using `String.fromCharCode(...)`? Instead of a fixed amount of characters. `String.fromCharCode(...)` encodes it into a valid utf-8 string which can one to five bytes of strings(depending on the number) and javascript already decodes the utf-8 strings by default.

Reply:
If numbers/records have fixed length, we can trivially find the offset for a particular entry:
```js
const base = 1 + 1 + (offsetLength + lengthLength) * index;
```
 If we use UTF-8, we will need to iterate over the entire string every time. It is just a tradeoff O(1) search time with more space vs. O(n) search time with a bit less space. In either case, space usage is roughly the same.